persistence-jpa/javadsl/src/main/java/com/lightbend/lagom/internal/javadsl/persistence/jpa/JpaReadSideImpl.java

```diff
@@ -168,15 +171,61 @@ private void updateOffset(EntityManager entityManager, Offset offset) {
                     asJavaIterable(offsetDao.updateOffsetQuery(OffsetAdapter.dslOffsetToOffset(offset)).statements());
 
             for (String statement : sqlStatements) {
+                log.debug("Updating offset to {} in JpaReadsideHandler {} with statement: {}",
+                        offset, readSideId, statement);
+                Query updateOffsetQuery = entityManager.createNativeQuery(statement);
                 // NOTE: The order of parameters here depends on the order chosen
-                // by Slick, based on the table definition in JdbcOffsetStore.
-                entityManager.createNativeQuery(statement)
-                        .setParameter(1, readSideId)
-                        .setParameter(2, tag.tag())
-                        .setParameter(3, sequenceOffset)
-                        .setParameter(4, timeUuidOffset)
-                        .executeUpdate();
+                // by Slick, based on the table definition in JdbcOffsetStore
+                // and the specific database profile in use.
+                JdbcProfile profile = getSlickDatabaseProfile();
+                if (profile instanceof PostgresDriver) {
+                    postgresqlBindUpdateOffsetQuery(updateOffsetQuery, sequenceOffset, timeUuidOffset);
+                } else {
+                    defaultBindUpdateOffsetQuery(updateOffsetQuery, sequenceOffset, timeUuidOffset);
+                }
```

Comment:
I suppose it could be, but it's already merged now! Think it's worth a follow-up issue?